src/sections/home-text-with-images.liquid

```diff
@@ -55,8 +55,11 @@
     color: #fff;
   }
   {% for block in section.blocks %}
-    .text-columns-with-images .block-{{ block.id }} .image-container::after {
-      opacity: {{ block.settings.image_overlay_opacity | times: 0.1 }};
+    .block-{{ block.id }} .image-container::after {
+      opacity: {{ section.settings.image_overlay_opacity | times: 0.1 }};
+    }
+    .block-{{ block.id }} .button {
+      background-color: {{ block.settings.background_color }};
```

Comment:
@rdnvol also could you please add conditions to check the emptiness of color_picker and opacity fields? For reference you may check `header.liquid` section